ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -294,28 +303,61 @@ module Wbinfo = struct
     | [] ->
         Error (parsing_ex args)
 
-  let domain_of_uname uname =
+  let domain_name_of ~target_name_type ~from_name =
     (*
-     * Get domain from domainified uname, which is formatted as NETBIOS\sAMAccountName
-     *
-     * convert NETBIOS name to domain by wbinfo --domain-info
+     * The domain name can be a normal domain name or netbios name
+     * Query the target domain name from one provided name
      *
      * example output:
      * Name              : UCC
      * Alt_Name          : ucc.local
      * SID               : S-1-5-21-2850064427-2368465266-4270348630
      * *)
+    let args = ["--domain-info"; from_name] in
+    let* stdout = call_wbinfo args in
+    let key = match target_name_type with
+    | Name -> "Alt_Name"
+    | NetbiosName -> "Name" in
+    try
+      Ok(Xapi_cmd_result.of_output ~sep:':' ~key stdout)
+    with _ -> Error (parsing_ex args)
+
+  let is_domain_netbios_valid domain_netbios =
+    let min_valid_domain_length = 1 in
+    match domain_name_of ~target_name_type:Name ~from_name:domain_netbios with
+    | Ok name -> String.length name >= min_valid_domain_length
+    | Error _ -> false
+
+  let domain_of_uname uname =
     match String.split_on_char '\\' uname with
-    | [domain_netbios; _] -> (
-        let args = ["--domain-info"; domain_netbios] in
-        let* stdout = call_wbinfo args in
-        try
-          Ok (domain_netbios, Xapi_cmd_result.of_output ~sep:':' ~key:"Alt_Name" stdout)
-        with _ -> Error (parsing_ex args)
-      )
+    | [domain_netbios; _] ->
+      let* domain_name = domain_name_of ~target_name_type:Name ~from_name:domain_netbios in
+      Ok (domain_netbios, domain_name)
     | _ ->
         Error (generic_ex "Invalid domain user name %s" uname)
 
+
+  let all_domain_netbis() =
```

Comment:
i'm not sure what the difference between netbis and netbios is, is this just a typo?